core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -621,14 +621,22 @@ object SparkSubmit {
     if (isKubernetesCluster) {
       childMainClass = "org.apache.spark.deploy.kubernetes.submit.Client"
       if (args.isPython) {
+        childArgs += "--py-file"
         childArgs += args.primaryResource
+        childArgs += "--main-class"
         childArgs += "org.apache.spark.deploy.PythonRunner"
+        childArgs += "--other-py-files"
         childArgs += args.pyFiles
       } else {
+        childArgs += "--primary-java-resource"
```

Comment:
java and python resources can be resolved with just a check if it ends with `.py`,  why is it necessary to pass down extra information from SparkSubmit that isn't needed per say. SparkSubmit imo should have minimal things being passed into our childMainClass